x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/config.tsx

```diff
@@ -0,0 +1,819 @@
+  [FormMonitorType.SINGLE]: [MONITOR_DETAILS_STEP],
+  [FormMonitorType.HTTP]: [MONITOR_DETAILS_STEP],
+  [FormMonitorType.ICMP]: [MONITOR_DETAILS_STEP],
+  [FormMonitorType.TCP]: [MONITOR_DETAILS_STEP],
+};
+
+export const MONITOR_TYPE_CONFIG = {
+  [FormMonitorType.MULTISTEP]: {
+    id: 'syntheticsMonitorTypeMultiStep',
+    label: 'Multistep',
+    value: FormMonitorType.MULTISTEP,
+    descriptionTitle: 'Multistep Browser Journey',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'videoPlayer',
+    beta: true,
+  },
+  [FormMonitorType.SINGLE]: {
+    id: 'syntheticsMonitorTypeSingle',
+    label: 'Single Page',
+    value: FormMonitorType.SINGLE,
+    descriptionTitle: 'Single Page Browser Test',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'videoPlayer',
+    beta: true,
+  },
+  [FormMonitorType.HTTP]: {
+    id: 'syntheticsMonitorTypeHTTP',
+    label: 'HTTP Ping',
+    value: FormMonitorType.HTTP,
+    descriptionTitle: 'HTTP Ping',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'online',
+    beta: false,
+  },
+  [FormMonitorType.ICMP]: {
+    id: 'syntheticsMonitorTypeICMP',
+    label: 'ICMP Ping',
+    value: FormMonitorType.ICMP,
+    descriptionTitle: 'ICMP Ping',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'online',
+    beta: false,
+  },
+  [FormMonitorType.TCP]: {
+    id: 'syntheticsMonitorTypeTCP',
+    label: 'TCP Ping',
+    value: FormMonitorType.TCP,
+    descriptionTitle: 'TCP Ping',
+    description:
+      'A lightweight API check to validate the availability of a web service or endpoint.',
+    link: '#',
+    icon: 'online',
+    beta: false,
+  },
+};
+
+export const FIELD: Record<string, FieldMeta> = {
+  [ConfigKey.MONITOR_TYPE]: {
+    fieldKey: 'formMonitorType',
+    required: true,
+    component: MonitorTypeRadioGroup,
+    ariaLabel: 'Monitor Type',
+    controlled: true,
+    props: ({ value, reset }) => ({
+      onChange: (_: string, monitorType: FormMonitorType) => {
+        const defaultFields = DEFAULT_FORM_FIELDS[monitorType];
+        reset(defaultFields);
+      },
+      selectedOption: value,
+      options: Object.values(MONITOR_TYPE_CONFIG),
+    }),
+    validation: () => ({
+      required: true,
+    }),
+  },
+  [`${ConfigKey.URLS}__single`]: {
+    fieldKey: ConfigKey.URLS,
+    required: true,
+    component: EuiFieldText,
+    label: 'Website URL',
+    helpText: 'For example, https://www.elastic.co.',
+  },
+  [`${ConfigKey.URLS}__http`]: {
+    fieldKey: ConfigKey.URLS,
+    required: true,
+    component: EuiFieldText,
+    label: 'URL',
+    helpText: 'For example, your service endpoint.',
```

Comment:
This should include required validation. I guess if I have the top level required prop (which is used to determine whether or not to show the optional tag), I can include the required validation automatically.